waltid-services/waltid-verifier-api/src/main/kotlin/id/walt/verifier/VerifierApi.kt

```diff
@@ -260,46 +266,51 @@ fun Application.verfierApi() {
-                            val state = session.walletInitiatedAuthState
-                            val code = UUID().toString()
-                            context.respondRedirect("openid://?code=$code&state=$state")
-                        } else {
-                            call.respond(HttpStatusCode.OK, it)
-                        }
-                    }.onFailure {
-                        logger.debug(it) { "Verification failed ($it)" }
-                        var errorDescription = it.localizedMessage
-                        logger.error { "Error: $errorDescription" }
-                        if (sessionId != null) {
-                            val session = verificationUseCase.getSession(sessionId)
+                runCatching {
+                    logger.info { "POST verify/state" }
+                    val sessionId = call.parameters["state"]
+                    logger.info { "State: $sessionId" }
+                    verificationUseCase.verify(sessionId, context.request.call.receiveParameters().toMap())
+                        .onSuccess {
+                            val session = verificationUseCase.getSession(sessionId!!)
                             if (session.walletInitiatedAuthState != null) {
                                 val state = session.walletInitiatedAuthState
-                                when (it.localizedMessage) {
-                                    "Verification policies did not succeed: expired" -> errorDescription =
-                                        "<\$presentation_submission.descriptor_map[x].id> is expired"
-
-                                    "Verification policies did not succeed: not-before" -> errorDescription =
-                                        "<\$presentation_submission.descriptor_map[x].id> is not yet valid"
-
-                                    "Verification policies did not succeed: revoked_status_list" -> errorDescription =
-                                        "<\$presentation_submission.descriptor_map[x].id> is revoked"
+                                val code = UUID().toString()
+                                context.respondRedirect("openid://?code=$code&state=$state")
+                            } else {
+                                call.respond(HttpStatusCode.OK, it)
+                            }
+                        }.onFailure {
+                            logger.debug(it) { "Verification failed ($it)" }
+                            var errorDescription = it.localizedMessage
+                            logger.error { "Error: $errorDescription" }
+                            if (sessionId != null) {
+                                val session = verificationUseCase.getSession(sessionId)
+                                if (session.walletInitiatedAuthState != null) {
+                                    val state = session.walletInitiatedAuthState
+                                    when (it.localizedMessage) {
```

Comment:
not sure checking the `localizedMessage` is a good approach
but anyway, maybe the error description could be extracted to separate method, e.g.:
```kotlin
private fun getErrorDescription(it: Throwable): String? = when (it.localizedMessage) {
    "Verification policies did not succeed: expired" ->
        "<\$presentation_submission.descriptor_map[x].id> is expired"

    "Verification policies did not succeed: not-before" ->
        "<\$presentation_submission.descriptor_map[x].id> is not yet valid"

    "Verification policies did not succeed: revoked_status_list" ->
        "<\$presentation_submission.descriptor_map[x].id> is revoked"

    else -> null
}
```
then have it called by `processError`:
```kotlin
private suspend fun PipelineContext<Unit, ApplicationCall>.processError(
    sessionId: String, exception: Throwable
) {
    val session = verificationUseCase.getSession(sessionId)
    if (session.walletInitiatedAuthState != null) {
        val state = session.walletInitiatedAuthState
        context.respondRedirect(
            "openid://?state=$state&error=invalid_request&error_description=${getErrorDescription(exception)}"
        )
    }
}
```
then extract the failure case into:
```kotlin
private suspend fun PipelineContext<Unit, ApplicationCall>.processVerificationFailureResult(
    it: Throwable,
    sessionId: String?
) {
    logger.debug(it) { "Verification failed ($it)" }
    val errorDescription = it.localizedMessage
    logger.error { "Error: $errorDescription" }
    if (sessionId != null) {
        processError(sessionId, it)
    } else {
        logger.error(it) { "/verify error: $errorDescription" }
        call.respond(HttpStatusCode.BadRequest, errorDescription)
    }
}
```
and the same for success:
```kotlin
private suspend fun PipelineContext<Unit, ApplicationCall>.processVerificationSuccessResult(
    sessionId: String?,
    it: String
) {
    val session = verificationUseCase.getSession(sessionId!!)
    if (session.walletInitiatedAuthState != null) {
        val state = session.walletInitiatedAuthState
        val code = UUID().toString()
        context.respondRedirect("openid://?code=$code&state=$state")
    } else {
        call.respond(HttpStatusCode.OK, it)
    }
}
```
then the verify handling would look like:
```kotlin
verificationUseCase.verify(sessionId, context.request.call.receiveParameters().toMap())
    .onSuccess {
        processVerificationSuccessResult(sessionId, it)
    }.onFailure {
        processVerificationFailureResult(it, sessionId)
    }.also {
        sessionId?.run { verificationUseCase.notifySubscribers(this) }
    }
```